### DISCUSSION TITLE
[BP#51] Suppression de "Privilégier les fonctions anonymes"

### DISCUSSION THREAD

**Discussion by ACTLEM** [cnumr/best-practices#61] [Bonnes pratiques] [answered]
En plus du fait que cela ne fait pas réellement gagner des ressources, cela engendre souvent un manque de lisibilité et de maintenabilité de manière.
D'autre part, mettre une fonction anonyme dans une boucle aura largement l'effet inverse escompté. Le rapport risque/bénéfice de cette bonne pratique me pousse donc à penser qu'il est souhaitable de la supprimer.

**Comment by didierfred** [comment]
Avec un [script de test](https://github.com/cnumr/best-practices/files/7405018/test_inline.zip) , je mesure sur mon Pc 
-  6ms de temps de calcul en plus  pour 10 Millions d'appel 
-  60 ms de temps de calcul en plus  pour 100 Millions d'appel 
-  1s de temps de calcul en plus pour 1 Milliard d'appel 

Dans le cas de calcul intensif on peut gagner mais pour les cas courant  ca n'a que peut d'impact .

Au vu effectivement de l'inconvenient sur l'aspect lisibilité  , cette bonne pratique peut être supprimée

**Comment by ACTLEM** [comment, marked as answer]
https://github.com/cnumr/best-practices/issues/126
